src/wallet/wallet.cpp

```diff
@@ -2297,8 +2474,10 @@ bool CWallet::CreateTransaction(const vector<CRecipient> &vecSend,
     {
         LOCK(cs_wallet);
         {
-            nFeeRet = 0;
-            // Start with no fee and loop until there is enough fee
+            CAmount nFeeNeeded = 0;
+            // Estimate base fee from an approx minimum size tx
+            nFeeRet = GetMinimumFee(MIN_BYTES_IN_TX, nTxConfirmTarget, mempool);
+            // Loop until there is enough fee
```

Comment:
Why we need to modify also the old way of selecting coins from the wallet?

Reply:
I don't understand your question.  The purpose of this PR is to select coins from the wallet more efficiently than the old way.

Reply:
I thought we implement the new way in `CWallet::SelectCoinsBU` and keep `CWallet::SelectCoins` as a backup solution. I could have thought that a bunch of changes to the latter have to be done, but not as substantial as it seems to me by looking at the PR changes.

Reply:
SelectCoins is dead code, I'll just remove it.